### ISSUE TITLE
Local branch is ahead of master. Automatic update not possible.

### ISSUE THREAD

**Issue by rafaelrosars** [SickChill/SickChill#6681]
Branch/Commit: master/99336df163c7e776c7931d864e84d1f0f83afe4d
OS: Raspibian
Browser: Safari
What you did: Try to update
What happened: Local branch is ahead of master. Automatic update not possible.
What you expected: update
Logs:
```
2020-09-05 19:36:01 WARNING WEBSERVER-HOME :: Local branch is ahead of master. Automatic update not possible. 2020-09-05 19:35:58 INFO WEBSERVER-HOME :: Checking for updates using GIT
```

**Comment by miigotu** [comment]
Have you tried reading the news in the past week?

**Comment by miigotu** [comment]
Python3 code has been merged into master. I intentionally created a continuity break in the commit history with a force push so that users would not be automatically updated to python3 code while their service files were pointing to python2 (causing it to not even start up)

This way, users can stay on python2 until they manually intervene and get everything working on python3 because the updater will refuse to update.
This also gives time for 3rd party maintainers to update their respective apps for specific devices.

Synology is being updated as we speak, there is a new Windows installer already, git will still work (but it is no longer the preferred install method)

Ideally we can get all users switched over to a pure pip install or using docker and eliminate a ton of issues, because everyone will have a system operating the same way.

If someone needs some specific help feel free to ask.

If anyone would like to help updating the wiki with new information, as well as a new page specifically for this situation that would be appreciated.

**Comment by miigotu** [comment]
Also, please forgive me if I seem short over the next few days. I will be helping hundreds of users and trying to get all of our packages updated while taking care of a 1 month old  baby :neckbeard:

**Comment by augur42** [comment]
Well I hadn't run SickChill in a week and imagine my surprise when it failed to work and I found your pointer to the news with an age of 48mins. As glad as I am for the move to python 3 I did not have a straightforward upgrade using the SickChillInstaller you just posted, I'll post about the issues there. I hope the errors were of my own making and as daft as it sounds I couldn't find the discord url anywhere, not even in the Wiki or FAQ.

PS congratulations on the baby.

**Comment by snapperhead** [comment]
current window installer downloads a portable version of git and installs to a temp folder then tries to run from C:\SickChill\git and fails with file not found.

**Comment by miigotu** [comment]
@snapperhead windows installer is fixed 0.5.7

**Comment by snapperhead** [comment]
> @snapperhead windows installer is fixed 0.5.7
wont start after install i dont think python install is ending up in sickchill dir as it looks like thats where its expected to be.

**Comment by augur42** [comment]
> 
> 
> > @snapperhead windows installer is fixed 0.5.7
> > wont start after install i dont think python install is ending up in sickchill dir as it looks like thats where its expected to be.

Have a look here. 
https://github.com/SickChill/SickChillInstaller/issues/27
This was for the first updated installer I 'used' 0.5.4
It might be the python path or it might be that the nssm service needs editing.
When you figure out which it is post an issue on the SickChillInstaller page 
https://github.com/SickChill/SickChillInstaller

**Comment by snapperhead** [comment]
> It might be the python path or it might be that the nssm service needs editing.
> When you figure out which it is please post an issue on the SickChillInstaller page
> https://github.com/SickChill/SickChillInstaller

turns out it didn't like my existing install of python3 once i uninstalled python3 and re-ran script everything is up and running now.

**Comment by havoc77** [comment]
Just wanted to confirm that Installer v0.5.7 worked for me no problems.  I am running on Windows 10 v2004

Here is my full process.

1. Stop SickChill service.
2. Backup everything in C:\SickChill to C:\SickChill.old
3. Add / Remove Programs and uninstall SickChill (Do not delete your database and settings)
4. I confirmed the only thing remaining in C:\SickChill was my Data folder
5. Run SickChillInstaller.exe v0.5.7

The installer restarted my service and I confirmed it is running again, and I confirmed that embedded Git and Python are in C:\SickChill

**Comment by augur42** [comment]
I've been testing 0.5.7 with miigotu on discord and I too failed to uninstall my existing python 3 which causing the install to fail. I then doubled down and uninstalled SickChill which broke python so I had to run a repair before uninstalling. 

Also there is one current gotcha wrt the image cache location changing (posters and banners and backgrounds).
old location was 
C:\SickChill\Data\gui\slick\cache\images
new location is
C:\SickChill\SickChill\sickchill\gui\slick\cache\images
The solution is to either copy the contents of \images and \images\thumbnails to the new location or run a Mass Update on all shows to redownload them.

**Comment by havoc77** [comment]
> Also there is one current gotcha wrt the image cache location changing (posters and banners and backgrounds).

Thanks for this I had not noticed it yet.

It made more sense to me to have this in the old Data folder and not under the repository folder....  I will move mine, but can any of the Devs confirm if this is intentional?

**Comment by miigotu** [comment]
No it is supposed to be symlinked into the data folder, but I'm exhausted so I'll get to it when I can lol

**Comment by bartmans** [comment]
A baby always comes first! Enjoy him or her, sleep (if you are allowed) and SC comes last ...

**Comment by waltermaldonado** [comment]
> Just wanted to confirm that Installer v0.5.7 worked for me no problems. I am running on Windows 10 v2004
> 
> Here is my full process.
> 
> 1. Stop SickChill service.
> 2. Backup everything in C:\SickChill to C:\SickChill.old
> 3. Add / Remove Programs and uninstall SickChill (Do not delete your database and settings)
> 4. I confirmed the only thing remaining in C:\SickChill was my Data folder
> 5. Run SickChillInstaller.exe v0.5.7
> 
> The installer restarted my service and I confirmed it is running again, and I confirmed that embedded Git and Python are in C:\SickChill

Just follow those steps but no folder C:\SickChill\Python3 was created, because I already have Python 3.8 installed on my system (I think) and I cannot uninstall it because I have other things that rely on this default installation. I had to copy the folder from C:\Python38 to C:\SickChill\ and rename to C:\SickChill\Python3 to get the service to start.

**Comment by ebaydan777** [comment]
when you say synology is being updated as we speak, do you mean Docker? And will it be as simple as updating docker itself or is there more we need to do with linuxserver-sickrage? Please help those like myself with Docker/linuxserver-sickrage and what we need to do instructions wise somewhere. I myself have python3 installed on my synology but not sure what else needs to be done. thank you!

**Comment by miigotu** [comment]
@ebaydan777 https://github.com/SickChill/SickChill/wiki/Docker
For regular syno installs, you can use sherpa or https://github.com/BenjV/SYNO-packages

**Comment by havoc77** [comment]
Love your work @miigotu 
Get some sleep!!!

**Comment by Tiwas** [comment]
@miigotu FYI - I tried the windows installer yesterday, but it failed because I already had SC installed. There was no uninstaller. Just wanted to mention it since you're referring to the installer earlier.

**Comment by havoc77** [comment]
@Tiwas You did not have it in your Windows Settings -> Apps -> Apps & Features list?
I can find it there and click the Uninstall button.

On my current system I have only ever installed via some form of the SickChillInstaller.exe - maybe you are different.

**Comment by spoelly** [comment]
> @ebaydan777 https://github.com/SickChill/SickChill/wiki/Docker
> For regular syno installs, you can use https://github.com/BenjV/SYNO-packages

I am running 1.1 version of BenV package currently on my syno, i would like to update to the pyton3 version but the installer won't let me.

**Comment by Tiwas** [comment]
> @Tiwas You did not have it in your Windows Settings -> Apps -> Apps & Features list?
> I can find it there and click the Uninstall button.
> 
> On my current system I have only ever installed via some form of the SickChillInstaller.exe - maybe you are different.

@havoc77 It's in the list, but it refers to an uninstall executable that doesn't exist.

**Comment by FGOD1983** [comment]
having issues getting the new installer to work:

```
pip install sickchill
Defaulting to user installation because normal site-packages is not writeable
ERROR: Could not find a version that satisfies the requirement sickchill (from versions: none)
ERROR: No matching distribution found for sickchill
```

I checked my python3 version and it set to 3.5.2 which should work...

**Comment by havoc77** [comment]
I have no idea about Python version compatibility and have never used pip... but but in the news page of my SickChill server from 2020-08-06 it says:
> Make sure you have python3.5+ installed on your system. There are tons of changes already, all for the better.

A newer post from 2020-08-31 says 
> I am going to do my best to wait a week from today before I merge develop to master.
> When this happens, if you are using docker, pip, etc everything will be seemless.
> 
> For everyone else, make a backup first, as you may need to restore the backup after manually updating and you do not want to lose any data.
> 
> If you are using git install, your updater is going to fail/error and you will need to make sure you have python3.6+
> installed and manually update your service file to use python3, and then do a git pull.
> Source installs will need to have python3.6+ installed and update your service file, and copy in the new source manually.
> A windows installer update will be released, that uses pip as the base install. Windows will need to uninstall and reinstall with the new installer. 

The newer post does mention python3.6+ though, if that helps?

**Comment by FGOD1983** [comment]
updated to python3 to 3.8.5, and still not working

**Comment by miigotu** [comment]
@FGOD1983 your pip is still using python2.7 (`pip -V` will show this)

Check `pip3 -V`
```bash
pip3 install -U sickchill
```

**Comment by Manfred73** [comment]
> @ebaydan777 https://github.com/SickChill/SickChill/wiki/Docker
> For regular syno installs, you can use https://github.com/BenjV/SYNO-packages

I have BenjV v1.2 installed and Python 3.7.7-12. How do I update SickChill manually so it uses this version of Python (where is the sickchill.service file located on Synology)?

**Comment by miigotu** [comment]
@Manfred73 BenJV took the python3 package down to fix it. You don't do anything manually on syno
https://github.com/SickChill/SickChill/issues/6697

**Comment by Manfred73** [comment]
> @Manfred73 BenJV took the python3 package down to fix it. You don't do anything manually on syno
> #6697

So when this is fixed, all I need to do is install the package? And this will then install new version of sickchill with Python3 and also convert the old db?

**Comment by miigotu** [comment]
The db does not need converted. It will install SC with python3 and use your existing data (make a backup just in case of your db and config.ini ALWAYS)
You might need to uninstall the old package and install the new one

**Comment by francisconora** [comment]
So to the users who were still using the "sickbeard-custom Method" for Synology, this means that I've must:
- do the backup from Sickchill
- shutdown the process (sickbeard-custom)
- install the BenJV package (when fixed and available)
- restore from the previous backup

Please do correct if I've said anything wrong.

Thank you in advance

**Comment by BKSteve** [comment]
> 
> 
> So to the users who were still using the "sickbeard-custom Method" for Synology, this means that I've must:
> 
>     * do the backup from Sickchill
> 
>     * shutdown the process (sickbeard-custom)
> 
>     * install the BenJV package (when fixed and available)
> 
>     * restore from the previous backup
> 
> 
> Please do correct if I've said anything wrong.
> 
> Thank you in advance

If you have the older BenjV Python2 noarch spk installed it needs to be uninstalled to allow installation the Sickchill Python3 spk

**Comment by FGOD1983** [comment]
> @FGOD1983 your pip is still using python2.7 (`pip -V` will show this)
> 
> Check `pip3 -V`
> 
> ```shell
> pip3 install -U sickchill
> ```

seems even pip3 and pip3.8 are using my pyhton 2.7 :/ time to do some troubleshooting on that...

**Comment by bartmans** [comment]
I updated the BenjV package like desribed above on a Synology. When I add a series, the owner and group are 'root' and not sc-sickchill.

**Comment by cbinder4** [comment]
> No it is supposed to be symlinked into the data folder, but I'm exhausted so I'll get to it when I can 

So for now we should not worry about moving images to another path as this will be fixed in a coming update?  

I also wanted to say, thanks to the info in this thread I was able to uninstall my Python installs and my old manual SickRage/SickChill install and use the Windows installer to update and get right back up and running.  Thanks, all!

**Comment by miigotu** [comment]
@cbinder4 you can move them, it is a bit different on windows and I was tired. Either you copy them or they will be re-cached in 24 hours.

**Comment by cbinder4** [comment]
Great, thanks.  I can wait for them to re-cache.  If waiting 24 hours for some images was the only issue I had moving from the original manual install and old Python to the new Windows installer and Python 3, then I say job well done.  My migration experience was nearly seemless.... all I really had to do was uninstall the manual installs and then after the new installer was done I updated the credentials on the service and restarted and seemed to be good to go.  Pretty painless in my opinion, so thanks for all of your work!

**Comment by glenviewjeff** [comment]
> Just wanted to confirm that Installer v0.5.7 worked for me no problems. I am running on Windows 10 v2004
> 
> Here is my full process.
> 
> 1. Stop SickChill service.
> 2. Backup everything in C:\SickChill to C:\SickChill.old
> 3. Add / Remove Programs and uninstall SickChill (Do not delete your database and settings)
> 4. I confirmed the only thing remaining in C:\SickChill was my Data folder
> 5. Run SickChillInstaller.exe v0.5.7
> 
> The installer restarted my service and I confirmed it is running again, and I confirmed that embedded Git and Python are in C:\SickChill

Tried this, appears to have worked other than the fact that all artwork has disappeared. Googling didn't help. Did others see this? Should I file a new issue?

**Comment by miigotu** [comment]
https://github.com/SickChill/SickChill/issues/6681#issuecomment-689035154

**Comment by richarda123** [comment]
> > So to the users who were still using the "sickbeard-custom Method" for Synology, this means that I've must:
> > ```
> > * do the backup from Sickchill
> > 
> > * shutdown the process (sickbeard-custom)
> > 
> > * install the BenJV package (when fixed and available)
> > 
> > * restore from the previous backup
> > ```
> > 
> > 
> > Please do correct if I've said anything wrong.
> > Thank you in advance
> 
> Synology DSM.
> If you have the older BenjV Python2 noarch spk installed it needs to be uninstalled to allow installation the Sickchill Python3 spk

Hello @BKSteve ,

Sorry to be such a P in the A but I don't know where to start. I've got SickChill installed on a Synology with the SickBeard Custom Package (20200804-8), Git (2.26.2-14) and Python (2.7.18-23) and since a few days I got a message: **Local branch is ahead of master. Automatic update not possible.**

To fix this, if I understand well I have to:

1. Save the SickChill DB: In SickChill go to Settings -> Backup & Restore -> Backup (is that it ?)
2. Save the config.ini file (where is it ?)
3. Do I need to stop the package SickBeard Custom in the Package Center ?
4. Go to https://github.com/BenjV/SYNO-packages and download the spk file "Medusa V1.0 Python3 noarch.spk" (is that it ?)
5. Install this package from the Package Center using the "Manual Install Button" ?
6. Then what ?

I feel I'm not far but I need help for the last steps.  Any help is welcome.

Thanks, Richard

**Comment by miigotu** [comment]
Literally none of that should be done.

**Comment by richarda123** [comment]
> Literally none of that should be done.

@miigotu THANKS ! (lol).
So what should I do ?
(my only concern is that **Local branch is ahead of master. Automatic update not possible** message. It tells me that something is wrong but what ?

**Comment by miigotu** [comment]
The message is just telling you that I intentionally made the updater error that way so that it did not automatically update you and break your install all by itself.

For SickChill on Synology sickbeard-custom is no longer a viable package. We run strictly under python3.6+ now. 

Stay with the current version you have until the new syno package is made/merged, directly on synocommunity.
https://github.com/SynoCommunity/spksrc/pull/4161

If that package is not merged soon, I will again provide a package right in our Org.

You could go there and like/comment on the PR, or contact Synology maintainers by other means (forums?) And ask for it to me merged.

**Comment by richarda123** [comment]
> The message is just telling you that I intentionally made the updater error that way so that it did not automatically update you and break your install all by itself.
> 
> For SickChill on Synology sickbeard-custom is no longer a viable package. We run strictly under python3.6+ now.
> 
> Stay with the current version you have until the new syno package is made/merged, directly on synocommunity.
> [SynoCommunity/spksrc#4161](https://github.com/SynoCommunity/spksrc/pull/4161)
> 
> If that package is not merged soon, I will again provide a package right in our Org.
> 
> You could go there and like/comment on the PR, or contact Synology maintainers by other means (forums?) And ask for it to me merged.

Thank you so much @miigotu . So I do nothing and I wait and I update the SickBeard Custom Package whenever it is available on Synocommunity. Crystal Clear !

**Comment by miigotu** [comment]
> Thank you so much @miigotu . So I do nothing and I wait and I update the SickBeard Custom Package whenever it is available on Synocommunity. Crystal Clear !

You will install the "sickchill" package by Nyaran once it is merged.

**Comment by Manfred73** [comment]
So what is Medusa. I noticed there is a new medusa v1.0 Pythons noarch.spk on BenjV github. I installed it manually. Is this the replacement for the old sickchill sich still users python 2.7? If I install this , how do I get my data from the old sickchill to medusa? Or is this what needs to be merged to a new official syno package that will update everything automatically?

**Comment by miigotu** [comment]
@Manfred73 no, it is not a replacement for SC. It is a different fork and we do not talk about competing forks here.

**Comment by richarda123** [comment]
> > Thank you so much @miigotu . So I do nothing and I wait and I update the SickBeard Custom Package whenever it is available on Synocommunity. Crystal Clear !
> 
> You will install the "sickchill" package by Nyaran once it is merged.
> 
> Make sure you make a backup from the old install, and restore it in the new install.

OK clear, thanks.
By The Way, could you please tell me how to backup the Settings (I believe the Config.ini). Where do I find this file ?

**Comment by BKSteve** [comment]
> OK clear, thanks.
> By The Way, could you please tell me how to backup the Settings (I believe the Config.ini). Where do I find this file ?

Use Sickchill backup function under settings.

**Comment by richarda123** [comment]
> > OK clear, thanks.
> > By The Way, could you please tell me how to backup the Settings (I believe the Config.ini). Where do I find this file ?
> 
> Use Sickchill backup function under settings.

I understand this saves the DB but we need to save the setting too don't we ?

**Comment by BKSteve** [comment]
> 
> I understand this saves the DB but we need to save the setting too don't we ?

It backs up the DB yes. It also backs up the 
sickchill.db
config.ini
failed.db
cache.db
And lots of image files.

**Comment by miigotu** [comment]
And movies.db if you have it =P

**Comment by miigotu** [comment]
Syno package built for testing: https://github.com/SynoCommunity/spksrc/pull/4161#issuecomment-690940177

**Comment by danyboy666** [comment]
> @FGOD1983 your pip is still using python2.7 (`pip -V` will show this)
> 
> Check `pip3 -V`
> 
> ```shell
> pip3 install -U sickchill
> ```
thanks for this I upgraded without issues

htpc@htpc-media-server:/opt/sickchill$ sudo service sickchill status
● sickchill.service - SickChill Daemon
     Loaded: loaded (/etc/systemd/system/sickchill.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2020-09-12 20:08:43 EDT; 5s ago
    Process: 2841513 ExecStart=/usr/bin/python3.8 /home/htpc/.local/bin/SickChill.py --daemon --nolaunch --datadir=/opt/sickchill (code=exited, status=0/SUCCESS)
      Tasks: 15 (limit: 9451)
     Memory: 108.3M
     CGroup: /system.slice/sickchill.service
             └─2841520 /usr/bin/python3.8 /home/htpc/.local/bin/SickChill.py --daemon --nolaunch --datadir=/opt/sickchill

Sep 12 20:08:40 htpc-media-server systemd[1]: Starting SickChill Daemon...
Sep 12 20:08:43 htpc-media-server systemd[1]: Started SickChill Daemon.
Sep 12 20:08:43 htpc-media-server python3.8[2841520]: /home/htpc/.local/lib/python3.8/site-packages/sickchill/locale
Sep 12 20:08:43 htpc-media-server python3.8[2841520]: /home/htpc/.local/lib/python3.8/site-packages/sickchill/locale
Sep 12 20:08:43 htpc-media-server python3.8[2841520]: /home/htpc/.local/lib/python3.8/site-packages/sickchill/locale

**Comment by FaiqH** [comment]
ok so I will have to find out how I installed sickchill on Syno and then do it again once you awesome guys complete it. thank you :)
I did the BenjV install last time and then again for dsm 6 (I had lost everything and had to do it again) I will look for the tutorial again.

**Comment by FaiqH** [comment]
> ok so I will have to find out how I installed sickchill on Syno and then do it again once you awesome guys complete it. thank you :)
> I did the BenjV install last time and then again for dsm 6 (I had lost everything and had to do it again) I will look for the tutorial again.

Oh I see the new spk, i will install that after the I update my python. Thank you.

**Comment by FaiqH** [comment]
> > ok so I will have to find out how I installed sickchill on Syno and then do it again once you awesome guys complete it. thank you :)
> > I did the BenjV install last time and then again for dsm 6 (I had lost everything and had to do it again) I will look for the tutorial again.
> 
> Oh I see the new spk, i will install that after the I update my python. Thank you.

Backup it up.
deleted older Chill
used spk mentioned above.
install and run (took time)
restored
profit
???

Thank you you guys, too this slow guy some time but I got it.
And congrats on the baby, I have a pandemic baby too. having kids in this time is tough. respect.

**Comment by FGOD1983** [comment]
oke, to fix my issues with python I took the easy way and reinstalled my whole server.

I'm getting to the point now that the install worked of sickchill and it is now installed in `.local/lib/python3.8/site-packages/sickchill` (?) after doing the `pip3 install -U sickchill` command... but I can't seem to find the service yet and what files do I need to take from my old install and move to the new location in order to get everything back to the old state?

**Comment by miigotu** [comment]
When you install with pip you have to edit one of the service files in contrib and install the service. It is not automatically set as a service (pip support is brand new for us)

**Comment by FGOD1983** [comment]
That was what i was thinking, as that is the normal way, but there is no contrib folder in the SickChill folder under `.local/lib/python3.8/site-packages/sickchill`
![Screenshot_20200914-203854](https://user-images.githubusercontent.com/18098873/93125078-e9873380-f6ca-11ea-914e-4bcba8fe8d85.png)

**Comment by miigotu** [comment]
@FGOD1983 Download it from https://github.com/SickChill/SickChill/tree/master/contrib/runscripts

**Comment by FGOD1983** [comment]
Thx, maybe I'm a bit slow here, but I still miss a lot of files in this folder... Should the whole thing still run from my old `/opt/sickchill` folder? Or from the new folder under the python directory? Because in the python directory I don't have the db files and SickChill.py as you can see in my picture...

Am I missing something? Should I copy over my old folder to the new python location?

**Comment by FGOD1983** [comment]
Oke, got it working again from the /opt/SickChill directory and restored my setup from there. 

But pure out of curiosity, what is the python SickChill folder used for, if the main data dir that still updates and holds the program/data files is used by the systemd service? 

Thx for the help and your patience btw 🙂

**Comment by miigotu** [comment]
You need to edit the systemd service to work with a pip install. By default SC uses a datadir in the home dir of the user running it (~/sickchill)

**Comment by miigotu** [comment]
@Manfred73 https://github.com/SynoCommunity/spksrc/pull/4161 is done they just haven't merged it yet.

**Comment by Manfred73** [comment]
For those interested in a generic update/restore script for some of these applications like SickChill that you have running on your Synology: https://github.com/Manfred73/SynologyBackupRestoreScripts.

**Comment by anb0s** [comment]
@miigotu So we need the new package available (merged) in SynoCommunity repo or can it be installed already manually? Thanks!

**Comment by Nyaran** [comment]
You can wait until de package is available on SynoCommunity, or install manually the package. Is provided on PR's first post.

**Comment by Manfred73** [comment]
> @miigotu So we need the new package available (merged) in SynoCommunity repo or can it be installed already manually? Thanks!

I did it manually last week. Got the spk from [BenjV github](https://github.com/BenjV/SYNO-packages/blob/master/SickChill%201.0%20Python%203%20noarch.spk).

And followed procedure:

- Backup it up old SickChill.
- Deleted older SickChill.
- Install new spk from BenjV mentioned above (do not start package initially).
- Restored from backup: had to rename sickbeard.db from backup to the new name sickchill.db.
- Started SickChill package.

**Comment by anb0s** [comment]
I did the upgrade now as average user ;) and may be it helps:
- create backup file with SickBeard Custom UI
- uninstall SickBeard Custom app with DSM UI
- download version from [PR#4161](https://github.com/SynoCommunity/spksrc/pull/4161) by @Nyaran (sickchill_noarch-all_20200920-0.tar.gz)
- extract sickchill_noarch-all_20200920-0.spk from above .tar.gz
- install spk with DSM UI (manual install) with no changes to the options
- open SickChill UI with provided link after SickChill was automatically started (port 8081)
- restore the backup file with SickChill UI
- [optional] verify with SSH that files where copied to `/volume1/@appstore/sickchill/var/data/restore`
- restart SickChill --> it shows shutdown succeeded, but stuck at start
- open the link with port 8083 (like was in the previous config.ini) and SickChill is there :)
- change the port to 8081 to have the valid link from DSM
- the database was restored with new name `sickchill.db` --> no need to rename from sickbeard.db manually

Now i see all the shows in SickChill but the posters (images) are not there, only this sickchill image instead:
![image](https://user-images.githubusercontent.com/95811/94447939-aa68e000-01aa-11eb-8cd3-4bab34ddff80.png)
How can i solve this? Thanks!

**Comment by miigotu** [comment]
> > @miigotu So we need the new package available (merged) in SynoCommunity repo or can it be installed already manually? Thanks!
> 
> I did it manually last week. Got the spk from [BenjV github](https://github.com/BenjV/SYNO-packages/blob/master/SickChill%201.0%20Python%203%20noarch.spk).
> 
> And followed procedure:
> 
> * Backup it up old SickChill.
> * Deleted older SickChill.
> * Install new spk from BenjV mentioned above (do not start package initially).
> * Restored from backup: had to rename sickbeard.db from backup to the new name sickchill.db.
> * Started SickChill package.

Do not use benjv package, or promote it here. We have an official package ready to be merged into synocommunity with real commit and change history.

**Comment by miigotu** [comment]
@anb0s force refresh on all shows or wait for the show updater to run. All images will be back within 24 hours.

**Comment by anb0s** [comment]
@miigotu thanks, "Mass update"--> select "Update" for all shows --> "Submit" and  now images are updated :)

**Comment by patdavid** [comment]
Is there a documented procedure for properly updating from git?  I'm on `master` right now and still seeing the update message.

**Comment by MystikIncarnate** [comment]
I had this problem; I went the manual method because I'm installed directly into Linux  (Ubuntu)  My process was simply:
- backup system using WebUI backup method
- stop SickChill
- uninstall SickChill using method documented in Wiki (moved install folder for additional backup in case of stuff going sideways)
- installed using manual method (also in Wiki)
- once started, restore from backup (SickChill required a restart)
- small exception: restart failed from web - sickchill restarted, but webUI was not responsive, required restarting again from Command line (SSH in my case).
- mass update meta data

Afterwards, everything is working normally and as expected.

I'll note that I did this because the old version I was running would not update metadata for new episodes; this has been an ongoing problem that I haven't found time to do a root-cause analysis on; however, once updated, one of the problematic series, that never seemed to update metadata automatically, was able to update metadata; though it was manual in this case, it's a good sign.

Thanks everyone who posted in this bug, I hope my comment helps someone else moving forward.

**Comment by patdavid** [comment]
I apparently lost my previous post somewhere, trying again.

I wanted to chime in and thank @MystikIncarnate for the nudge in the right direction.
In the end, backup all of my settings and just pulling a fresh version from git solved my problems.
Luckily the restore process is easy-peasy so once I had the new install using Python 3 I just had to restore my old settings.

Thanks again for all the work!

**Comment by DidierLoiseau** [comment]
@anb0s @miigotu I noticed the files are actually part of the backup, just probably not where SickChill expects them (in `/SickBeard/gui/slick/cache/`). I restored them manually and they immediately appeared in the UI.

**Comment by jonaspaulo** [comment]
Hi, did the upgrade fine on synology by manually installing the package but getting this error on the logs: 2020-10-15 11:47:03 Unable to open /root: PermissionError(13, 'Permission denied') / [Errno 13] Permission denied: '/root'

Despite the fact that I have sc-download with read/write everywhere
Thanks

**Comment by miigotu** [comment]
> Hi, did the upgrade fine on synology by manually installing the package but getting this error on the logs: 2020-10-15 11:47:03 Unable to open /root: PermissionError(13, 'Permission denied') / [Errno 13] Permission denied: '/root'
> 
> Despite the fact that I have sc-download with read/write everywhere
> Thanks

Maybe you installed the wrong package? Did you use the one from the Nyarans's PR on synocommunity, right?

**Comment by jonaspaulo** [comment]
Yep I used the one from https://github.com/SynoCommunity/spksrc/pull/4161 
Everything seems to be working ok even with that warning.
I cleared and it didn't appear again

**Comment by rafaelrosars** [comment]
> Branch/Commit: master/99336df163c7e776c7931d864e84d1f0f83afe4d
> OS: Raspibian
> Browser: Safari
> What you did: Try to update
> What happened: Local branch is ahead of master. Automatic update not possible.
> What you expected: update
> Logs:
> 
> ```
> 2020-09-05 19:36:01 WARNING WEBSERVER-HOME :: Local branch is ahead of master. Automatic update not possible. 2020-09-05 19:35:58 INFO WEBSERVER-HOME :: Checking for updates using GIT
> ```

Sorry for my lack of knowledge. I researched before posting here but I didn't find a solution. I'm using Raspbian and I don't know how to update to use python3. I already looked and python is installed on my system, I just don't know how to make it work with sickchill

**Comment by patdavid** [comment]
@rafaelrosars - You can backup your current settings/db, remove SickChill, then re-install from scratch (and restore your backup).
Once you're on Py3, you should be able to use the updater again.

**Comment by MystikIncarnate** [comment]
@rafaelrosars : installing python3 on Raspbian is pretty easy, you can google a guide, that's the only prerequisite you really need.  for the sake of having things in once space, it should be as easy as running "sudo apt install python3", and the package manager should take care of the rest.  The command "python --version" should tell you what is currently default.

When you have python sorted, you can follow my procedure above and that will do the trick; in summary, back up your config from the sickchill web ui (configure (gear icon), backup) then rebase your sickchill install (wipe it first for cleanliness), then perform the initial setup and restore from backup.  It's pretty easy overall.  The fellows at sickchill have made it relatively painless to get done.  Good luck.

to be fair, I'm just restating what pat said, but more verbose for clarity.

**Comment by rmbleeker** [comment]
As someone who **has** read up on the subject, the above confuses me. It's my understanding that a git installation of SickChill is recommended **against** by the maintainers and as such unsupported. It's not even listed as an installation method on the wiki anymore, only Docker and pip installs. So I would like the SickChill maintainers to clarity on this please.

**Comment by miigotu** [comment]
> As someone who **has** read up on the subject, the above confuses me. It's my understanding that a git installation of SickChill is recommended **against** by the maintainers and as such unsupported. It's not even listed as an installation method on the wiki anymore, only Docker and pip installs. So I would like the SickChill maintainers to clarity on this please.

Huh? Git or pip is the preferred method. Docker uses the git method anyway. SOURCE installs are discouraged.

**Comment by TimKras** [comment]
What's the procedure to update SickChill on a Synology NAS.
I'm using this package:
![image](https://user-images.githubusercontent.com/5257192/97208097-0d09c780-17bb-11eb-9aa8-c3d0014ecbed.png)

**Comment by miigotu** [comment]
> What's the procedure to update SickChill on a Synology NAS.
> I'm using this package:
> ![image](https://user-images.githubusercontent.com/5257192/97208097-0d09c780-17bb-11eb-9aa8-c3d0014ecbed.png)
> 

https://github.com/SynoCommunity/spksrc/pull/4161

**Comment by rmbleeker** [comment]
> Huh? Git or pip is the preferred method. Docker uses the git method anyway. SOURCE installs are discouraged.

Then I completely misunderstood that part. I guess I'll use git then, but without Docker.

**Comment by enormouspoon** [comment]
> I had this problem; I went the manual method because I'm installed directly into Linux (Ubuntu) My process was simply:
> 
>     * backup system using WebUI backup method
> 
>     * stop SickChill
> 
>     * uninstall SickChill using method documented in Wiki (moved install folder for additional backup in case of stuff going sideways)
> 
>     * installed using manual method (also in Wiki)
> 
>     * once started, restore from backup (SickChill required a restart)
> 
>     * small exception: restart failed from web - sickchill restarted, but webUI was not responsive, required restarting again from Command line (SSH in my case).
> 
>     * mass update meta data
> 
> 
> Afterwards, everything is working normally and as expected.
> 
> I'll note that I did this because the old version I was running would not update metadata for new episodes; this has been an ongoing problem that I haven't found time to do a root-cause analysis on; however, once updated, one of the problematic series, that never seemed to update metadata automatically, was able to update metadata; though it was manual in this case, it's a good sign.
> 
> Thanks everyone who posted in this bug, I hope my comment helps someone else moving forward.

Thank you so much - this worked perfectly! I've been holding off on updating for so long but finally gave in. Following these steps made it straight forward and simple. Props!

**Comment by gniting** [comment]
I moved the old version of Sickchill and installed the new version in `/opt/sickchill` using:
`pip3 install -t /opt/sickchill`

The install went fine, but when I try and run /opt/sickchill/bin/SickChill.py, I get the following error:
```
Traceback (most recent call last):
  File "/opt/sickchill/bin/SickChill.py", line 14, in <module>
    import sickchill.start
ModuleNotFoundError: No module named 'sickchill'
```

I've tried futzing around with `PYTHONPATH` but was unable to resolve. Any hints?

**Comment by miigotu** [comment]
> Trying to upgrade to the latest version, running into problems.
> 
> I moved the old version of Sickchill and installed the new version in `/opt/sickchill` using:
> `pip3 install -t /opt/sickchill sickchill`
> 
> The install went fine, but when I try and run `/opt/sickchill/bin/SickChill.py`, I get the following error:
> 
> ```
> Traceback (most recent call last):
>   File "/opt/sickchill/bin/SickChill.py", line 14, in <module>
>     import sickchill.start
> ModuleNotFoundError: No module named 'sickchill'
> ```
> 
> I've tried futzing around with `PYTHONPATH` but was unable to resolve. Any hints?

You don't set a target that way or the wrapper doesn't work. Use the default path or set up a virtualenv instead.

```bash
<set up virtualenv, set WORKON_HOME in bashrc>

mkvirtualenv -p python3.8 sickchill
workon sickchill
pip3 install sickchill
```

**Comment by gniting** [comment]
> > Trying to upgrade to the latest version, running into problems.
> > I moved the old version of Sickchill and installed the new version in `/opt/sickchill` using:
> > `pip3 install -t /opt/sickchill sickchill`
> > The install went fine, but when I try and run `/opt/sickchill/bin/SickChill.py`, I get the following error:
> > ```
> > Traceback (most recent call last):
> >   File "/opt/sickchill/bin/SickChill.py", line 14, in <module>
> >     import sickchill.start
> > ModuleNotFoundError: No module named 'sickchill'
> > ```
> > 
> > 
> > I've tried futzing around with `PYTHONPATH` but was unable to resolve. Any hints?
> 
> You don't set a target that way or the wrapper doesn't work. Use the default path or set up a virtualenv instead.
> 
> ```shell
> <set up virtualenv, set WORKON_HOME in bashrc>
> 
> mkvirtualenv -p python3.8 sickchill
> workon sickchill
> pip3 install sickchill
> ```

Thanks for the hint. If I am using a startup script to get SickChill going and SickChill is running as a special user `sickchill` with no home directory, is there a diff set of recommendations?

**Comment by richarda123** [comment]
Hello, sorry to re-open the topic but I still don't understand what to do on my Synology.

I have SickChill running on the SickBeard Package from Diaoul on SynoCommunity (v 20200804-4) and still showing "Local branch is ahead of master. Automatic update not possible.".

I understand that a new SickChill specific package will be posted on SynoCommunity but there is nothing so far. 

What can/should the Synology users do ?

Thanks for the help

Richard

**Comment by miigotu** [comment]
@richarda123 diaoul's package does not work for newer sickchill. You have to back up, install the package from https://github.com/SynoCommunity/spksrc/pull/4161 and restore.

It should be merged already, but every time we make whatever changes they request they change the request and move the goal posts.

**Comment by richarda123** [comment]
OK, this is clear, thanks. However, when I go to this link: SynoCommunity/spksrc#4161 I search everywhere and I don't find the package I should download ans install on the Synology. I see the code but I can't find any package or any instruction on how to build it.

**Comment by miigotu** [comment]
It should automatically update, unless they keep having the source changed a thousand times. We have been at this with them for like 3 years trying to get a proper package in, and it is always this way.
